File: school/webium/april/27/c.cpp
#include <iostream>


int main()
{
    for (int num = 45'000'000; num <= 50'000'000; ++num)
    {
        int count_of_prime_divs = 0;
        for (int div = 1; div * div <= num; ++div)
        {
            if (num % div == 0)
            {
                count_of_prime_divs += div % 2 + (num / div) % 2;
                if (div * div == num && div % 2 == 1)
                    --count_of_prime_divs;
            }
            if (count_of_prime_divs > 5)
                break;
        }
        if (count_of_prime_divs == 5)
            std::cout << num << std::endl;
    }
    return 0;
}

File: school/webium/april/27/c.py
for num in range(45_000_000, 50_000_001):
    count_of_odd_divs = 0 if int(num ** 0.5) ** 2 != num else -1
    for div in range(1, int(num ** 0.5) + 1):
        if num % div == 0:
            count_of_odd_divs += (div % 2) + ((num // div) % 2)
        if count_of_odd_divs > 5:
            break
    else:
        if count_of_odd_divs == 5:
            print(num)
